pcdsdevices/epics_motor.py

```diff
@@ -735,6 +737,27 @@ def flag_is_cleared(value=None, **kwargs):
             status_wait(st, timeout=timeout)
         return st
 
+    @property
+    def md(self):
+        if self._md is None:
+            raise AttributeError('Device does not have an attached md, '
+                                 'and was likely not initialized from happi')
+        return self._md
+
+    @md.setter
+    def md(self, new_md):
+        """ initialize attributes when md is set """
+        self._md = new_md
+        self._extra = self._md.extraneous
+        self._id = self._extra['_id']
+        self._stageidentity = self._extra['stageidentity']
+        self._pvbase = self._extra['pvbase']
+        if self._stageidentity == "NEW":
+            print(f"This is a new stage, parameter configuration does not exist yet. Please manually configure {self._id}.")
+        else:
+            IMS._setup_and_check_pmgr()
+            self._pm.set_config(self._pvbase, self._stageidentity)
```

Comment:
Note that this will cause pmgr database changes every time anyone starts a new hutch python session, even from dev machines etc.
I'm not sure if this is a problem or not, but I thought I'd note it.

Reply:
I tested this and it looks like the changes only get applied when you are on the specific hutch machine that the change is meant to happen in.